File: src/main/java/net/frozenorb/potpvp/util/menu/pagination/PageButton.java
package net.frozenorb.potpvp.util.menu.pagination;

import net.frozenorb.potpvp.util.menu.Button;
import lombok.AllArgsConstructor;
import org.bukkit.Material;
import org.bukkit.entity.Player;
import org.bukkit.event.inventory.ClickType;

import java.util.ArrayList;
import java.util.List;

@AllArgsConstructor
public class PageButton extends Button {
    private int mod;
    private PaginatedMenu menu;

    @Override
    public void clicked(Player player,int i,ClickType clickType) {

        if (clickType == ClickType.RIGHT) {
            (new ViewAllPagesMenu(this.menu)).openMenu(player);
            playNeutral(player);
        } else if (this.hasNext(player)) {
            this.menu.modPage(player, this.mod);
            Button.playNeutral(player);
        } else {
            Button.playFail(player);
        }

    }

    private boolean hasNext(Player player) {

        final int pg = this.menu.getPage() + this.mod;

        return pg > 0 && this.menu.getPages(player) >= pg;
    }

    @Override
    public String getName(Player player) {

        if (!this.hasNext(player)) {
            return this.mod > 0 ? "§7Last page" : "§7First page";
        } else {
            return this.mod > 0 ? "§a⟶" : "§c⟵";
        }

    }

    @Override
    public List<String> getDescription(Player player) {
        return new ArrayList();
    }

    @Override
    public byte getDamageValue(Player player) {
        return (byte)(this.hasNext(player) ? 11 : 7);
    }

    @Override
    public Material getMaterial(Player player) {
        return Material.CARPET;
    }
}

File: src/main/kotlin/net/frozenorb/potpvp/kt/menu/pagination/PageButton.kt
package net.frozenorb.potpvp.kt.menu.pagination

import net.frozenorb.potpvp.kt.menu.Button
import org.bukkit.Material
import org.bukkit.entity.Player
import java.util.ArrayList
import org.bukkit.event.inventory.ClickType
import org.bukkit.inventory.InventoryView

class PageButton(private val mod: Int, private val menu: PaginatedMenu) : Button() {

    override fun clicked(player: Player, i: Int, clickType: ClickType, view: InventoryView) {
        when {
            clickType == ClickType.RIGHT -> {
                ViewAllPagesMenu(menu).openMenu(player)
                playNeutral(player)
            }
            hasNext(player) -> {
                menu.modPage(player, mod)
                playNeutral(player)
            }
            else -> playFail(player)
        }
    }

    private fun hasNext(player: Player): Boolean {
        val pg = menu.page + mod
        return pg > 0 && menu.getPages(player) >= pg
    }

    override fun getName(player: Player): String {
        if (!this.hasNext(player)) {
            return if (this.mod > 0) "§7Last page" else "§7First page"
        }

        val str = "(§e" + (menu.page + mod) + "/§e" + menu.getPages(player) + "§a)"
        return (if (this.mod > 0) "§a\u27f6" else "§c\u27f5") + str
    }

    override fun getDescription(player: Player): List<String> {
        return ArrayList()
    }

    override fun getDamageValue(player: Player): Byte {
        return (if (this.hasNext(player)) 11 else 7).toByte()
    }

    override fun getMaterial(player: Player): Material {
        return Material.CARPET
    }

}